src/growth.jl

```diff
@@ -36,10 +36,11 @@ end
 
 precalcrule(rule::ExponentialGrowth, data) = precalc_timestep(rule, data)
 
-@inline function applyrule(data, rule::ExponentialGrowth, population, I)
-    population > zero(population) || return zero(population)
-    intrinsicrate = get(data, rule.rate, I...)
-    @fastmath population * exp(intrinsicrate * rule.nsteps)
+@inline function applyrule(data, rule::ExponentialGrowth, N, I)
+    N > zero(N) || return zero(N)
+    r = get(data, rule.rate, I...) * rule.nsteps
+
+    return @fastmath N * exp(r)
```

Comment:
If you want this in canonical form, it should really be `N * exp(r*t)` where t is `rule.nsteps`. 

To solve this I would change `r` to `rt` so its is implied.

Reply:
lets go full canonical